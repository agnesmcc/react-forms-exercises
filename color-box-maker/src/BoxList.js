import React, { useState } from 'react';
import Box from './Box';

const BoxList = () => {
    const [boxes, setBoxes] = useState([{color: "red", width: 100, height: 100}]);
    return (
        <div>
            <h1>Box List</h1>
            {boxes.map((box) => 
                <Box key={box.id} {...box} />)}
        </div>
    )
}

export default BoxList
