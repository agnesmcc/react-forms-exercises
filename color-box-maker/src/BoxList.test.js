import { render, fireEvent, getAllByText } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', () => {
  const { getByText } = render(<BoxList />);
  const heading = getByText(/Box List/);
  expect(heading).toBeInTheDocument();
})

it('matches the snapshot', () => {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
}) 

it('removes a box when the x is clicked', () => {   
    const { getByText } = render(<BoxList />)
    const button = getByText('X');
    fireEvent.click(button);
    expect(button).not.toBeInTheDocument();
})

it('is able to create a box using the form', () => {
    const { getByText, getByLabelText, getAllByText} = render(<BoxList />)
    
    // Submit the form to create a new box
    const widthInput = getByLabelText(/Width/);
    fireEvent.change(widthInput, {target: {value: 100}});
    const heightInput = getByLabelText(/Height/);
    fireEvent.change(heightInput, {target: {value: 100}});
    const colorInput = getByLabelText(/Color/);
    fireEvent.change(colorInput, {target: {value: 'blue'}});
    const button = getByText('Add a new box');
    fireEvent.click(button);

    // Check that the new box has been created
    const boxes = getAllByText('X');
    expect(boxes.length).toBe(2);
})
