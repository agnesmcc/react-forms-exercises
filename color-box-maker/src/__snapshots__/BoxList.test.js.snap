// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Box List
    </h1>
    <div
      style="background-color: red; width: 100px; height: 100px;"
    >
      <button>
        X
      </button>
    </div>
    <div>
      <form>
        <div>
          <label
            for="width"
          >
            Width
          </label>
          <input
            id="width"
            name="width"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="height"
          >
            Height
          </label>
          <input
            id="height"
            name="height"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="color"
          >
            Color
          </label>
          <input
            id="color"
            name="color"
            type="text"
            value=""
          />
        </div>
        <button>
          Add a new box
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
